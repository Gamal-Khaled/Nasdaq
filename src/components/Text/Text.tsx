import React from "react";
import { Text as RNText, TextProps, StyleSheet } from "react-native";

import theme from "../../theme/theme";
import variants from "./TextVariants";

export interface Props extends TextProps {
  variant?: keyof typeof variants;
  secondary?: boolean;
}

const Text = ({ variant, secondary, style, ...props }: Props) => {
  const _style = StyleSheet.flatten([
    !!variant && variants[variant],
    secondary ? styles.secondaryStyle : styles.defaultStyle,
    style,
  ]);

  return <RNText {...props} style={_style} />;
};

export const styles = StyleSheet.create({
  defaultStyle: {
    color: theme.font,
  },
  secondaryStyle: {
    color: theme.fontSecondary,
  },
});

export default Text;
