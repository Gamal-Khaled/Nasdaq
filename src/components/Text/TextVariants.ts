import { StyleSheet } from "react-native";

const variants = StyleSheet.create({
  title: {
    fontWeight: "bold",
    fontSize: 16,
  },
  subtitle: {
    fontSize: 12,
  },
});

export default variants;
