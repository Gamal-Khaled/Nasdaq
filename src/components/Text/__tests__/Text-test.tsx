import { cleanup, render, screen } from "@testing-library/react-native";

import variants from "../TextVariants";
import Text from "../Text";

afterEach(cleanup);

describe("Testing 'Text' component", () => {
  test("Should render with default variant if no variant specified", () => {
    render(<Text>Test Text</Text>);

    expect(screen.toJSON()).toMatchSnapshot();
  });
  test("Should render with the specified variant", () => {
    render(
      <Text variant="subtitle" testID="testText">
        Test Text
      </Text>
    );

    expect(screen.getByTestId("testText")).toHaveStyle(variants.subtitle);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
