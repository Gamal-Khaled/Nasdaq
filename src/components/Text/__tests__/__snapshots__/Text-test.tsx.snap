// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing 'Text' component Should render with default variant if no variant specified 1`] = `
<Text
  style={
    {
      "color": "#fff",
    }
  }
>
  Test Text
</Text>
`;

exports[`Testing 'Text' component Should render with the specified variant 1`] = `
<Text
  style={
    {
      "color": "#fff",
      "fontSize": 12,
    }
  }
  testID="testText"
>
  Test Text
</Text>
`;
